siege_utilities/schema/migration_runner.py

```
@@ -0,0 +1,358 @@
+            log.info("schema runner: applied %s (%s)", migration.version, migration.description)
+        return applied
+
+    def apply_up_to(self, target_version: str) -> list[PendingMigration]:
+        """
+        Apply pending migrations in order, stopping once ``target_version`` is applied.
+
+        Raises ``ValueError`` if ``target_version`` is not on disk.
+        """
+        on_disk_versions = [m.version for m in self.discover_migrations()]
+        if target_version not in on_disk_versions:
+            raise ValueError(f"target_version {target_version!r} not found in migrations dir")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)